apps/meteor/ee/app/livechat-enterprise/server/api/lib/contacts.ts

```diff
@@ -0,0 +1,52 @@
+import type { ILivechatContact, IUser } from '@rocket.chat/core-typings';
+import { License } from '@rocket.chat/license';
+import { LivechatContacts, LivechatRooms, LivechatVisitors } from '@rocket.chat/models';
+
+import { Livechat } from '../../../../../../app/livechat/server/lib/LivechatTyped';
+import { i18n } from '../../../../../../server/lib/i18n';
+
+export async function changeContactBlockStatus({ contactId, block, visitorId }: { contactId: string; visitorId: string; block: boolean }) {
+	const contact = await LivechatContacts.findOneById<Pick<ILivechatContact, '_id' | 'channels'>>(contactId, {
+		projection: { _id: 1, channels: 1 },
+	});
+
+	if (!contact) {
+		throw new Error('error-contact-not-found');
+	}
+
+	if (!contact.channels) {
+		throw new Error('error-contact-has-no-channels');
+	}
+
+	const channelIndex = contact.channels?.findIndex((channel) => channel.visitorId === visitorId);
+
+	if (channelIndex === -1) {
+		throw new Error('error-channel-not-found');
+	}
+
+	contact.channels[channelIndex].blocked = block;
+
+	await LivechatContacts.updateOne({ _id: contactId }, { $set: { channels: contact.channels } });
+}
+
+export async function hasSingleContactLicense() {
```

Comment:
why is this function async? 👀 `License.hasModule` is not async either. Plans to add some async check here? otherwise, pls make it sync.